#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/WoodenBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"

WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation 0.6113883956325834 -0.2561951511652073 -0.7487110752521644 3.749613800684158
  position 0.27736003417123434 -0.16885736623363248 2.246917737668706
}
TexturedBackground {
}
TexturedBackgroundLight {
}
E-puck {
  hidden position_0_0 -3750.093888063526
  hidden position_0_1 -4783.41078548025
  hidden linearVelocity_0 -5.63076093835785e-05 -8.743875129577445e-05 -1.603469185918894e-05
  hidden angularVelocity_0 0.005076934342965228 -0.005465709212760497 2.226951921156575
  hidden rotation_1 0 -0.9999999999999999 0 5.31544498449227
  hidden rotation_2 0 1 0 4.37641859059528
  translation -0.4349261572905011 0.02927622998223059 -3.140952329192849e-05
  rotation 0.0006088276319672745 0.0002529652063776246 0.9999997826687359 -2.3597239369372733
  controller "epuck_avoid_collision"
}
WoodenBox {
  translation -0.3855152823882769 -0.06968571391421075 0.04990190000000003
  rotation 9.280726504155929e-16 -2.507461788611409e-16 1 0.010189563653324398
  size 0.1 0.1 0.1
  mass 0.5
}
WoodenBox {
  translation -0.37662755337058673 -0.23667761549105115 0.04990190000000001
  rotation 6.874227806534176e-17 -1.5375438005002625e-17 1 0.10118540179167886
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 0.5
}
WoodenBox {
  translation -0.31959538664446935 0.2971194830117864 0.04990190000000002
  rotation -4.1292364180731904e-17 2.1440228156139675e-16 1 0.050673095373671075
  name "wooden box(2)"
  size 0.1 0.1 0.1
  mass 0.5
}
Floor {
  size 1 1
}
Solid {
  translation -0.0420343 0.340893 0.03977728118705348
  rotation -0.012410796645036956 0.9999229713302521 0.00015340376545160388 1.5707963503276123
  children [
    DEF dumble Group {
      children [
        Transform {
          translation 0 0 0.08
          children [
            DEF pendul Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.04
                subdivision 2
              }
            }
          ]
        }
        Transform {
          translation 0 0 -0.08
          children [
            USE pendul
          ]
        }
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.01
          }
        }
      ]
    }
  ]
  name "solid(6)"
  boundingObject USE dumble
  physics Physics {
  }
  linearVelocity -5.0184757103134246e-18 -1.1705299814936248e-18 -1.7526303779774362e-16
  angularVelocity 2.5535539039417366e-17 -1.1654239111442087e-16 -4.856756143381433e-18
}
Solid {
  translation 0.2559363943254904 0.17654968290864134 0.04958907968091044
  rotation 0.6814944263010978 0.6700051337395101 -0.29437810326727587 2.2945666477662567
  children [
    DEF BALL_GEOMETRY Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Sphere {
        radius 0.05
        subdivision 2
      }
    }
  ]
  boundingObject USE BALL_GEOMETRY
  physics Physics {
  }
  linearVelocity 0.002146218538286004 -0.00700019785716102 1.86423663182278e-16
  angularVelocity 0.1400039571432204 0.042924370765720074 -0.023319376192168846
}
Solid {
  translation -0.04544131026948041 -0.13566791840724038 0.04958907968091044
  rotation 0.6069313975095073 0.31433539001432365 -0.7299503690674451 0.8365090961193502
  children [
    DEF BALL_GEOMETRY Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/mrayh/Downloads/red_brick_wall.jpg"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Sphere {
        radius 0.05
        subdivision 2
      }
    }
  ]
  name "solid(5)"
  boundingObject USE BALL_GEOMETRY
  physics Physics {
  }
  linearVelocity 0.0035841772691503617 -0.002581639501973484 1.90063000274705e-16
  angularVelocity 0.05163279003946968 0.07168354538300722 0.018511811490680246
}
Solid {
  translation -3.52654e-06 0.489999 0.0300002
  rotation -2.7999599999889585e-06 -2.1785999999914086e-07 0.9999999999960565 -3.1415853071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/mrayh/Downloads/0001.jpg"
            ""
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.02 0.05
      }
    }
  ]
  name "solid(1)"
}
Solid {
  translation 0.490001 3.4e-06 0.030005
  rotation 3.0178299999761977e-06 -2.5821099999796344e-06 -0.9999999999921128 1.57079
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/mrayh/Downloads/0001.jpg"
            ""
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.02 0.05
      }
    }
  ]
  name "solid(2)"
}
Solid {
  translation -0.499998 -1e-06 0.0299953
  rotation 3.0178299999761977e-06 -2.5821099999796344e-06 -0.9999999999921128 1.57079
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/mrayh/Downloads/0001.jpg"
            ""
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.02 0.05
      }
    }
  ]
  name "solid(3)"
}
Solid {
  translation 4.9e-06 -0.490001 0.0300001
  rotation -2.7999599999889585e-06 -2.1785999999914086e-07 0.9999999999960565 -3.1415853071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "C:/Users/mrayh/Downloads/0001.jpg"
            ""
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.02 0.05
      }
    }
  ]
  name "solid(4)"
}
Robot {
  translation 0.0025252595887356147 0.12420884807740935 0.049534894016143506
  rotation 0.003117919339381448 -0.00019185060011520064 -0.9999951208742673 1.5115008873666829
  children [
    DistanceSensor {
      translation -0.0323205 0.099999 0
      rotation -9.58983e-09 -7.35849e-09 1 1.83258
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_kiri"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.0316869 -0.100258 1.22771e-09
      rotation -9.58983e-09 -7.35849e-09 1 1.83258
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_belakang2"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.0308113 0.1 1.21084e-09
      rotation 9.58989e-09 1.24977e-08 -1 -1.3089853071795865
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_kanan"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.0332017 -0.100256 -1.68734e-11
      rotation 9.58989e-09 1.24977e-08 -1 -1.3089853071795865
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_belakang1"
      boundingObject USE sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 26.15998242092201
        anchor -0.0744446 0.06 -1.20791e-08
      }
      device [
        RotationalMotor {
          name "roda1"
        }
      ]
      endPoint Solid {
        translation -0.07444439587065498 0.05999999330546492 -1.7407632441635284e-06
        rotation 0.44096527250044865 -0.7817286387390353 -0.44096481018763833 1.8145913977355674
        children [
          DEF roda Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(3)"
        boundingObject USE roda
        physics DEF phisics_roda Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 13.36002742510578
        anchor -0.0744444 -0.07 7.65305e-06
      }
      device [
        RotationalMotor {
          name "roda2"
        }
      ]
      endPoint Solid {
        translation -0.07444442074023845 -0.06999997285728071 1.3979109986550125e-05
        rotation 0.36048810659711517 -0.8602889438846035 -0.36048752826129865 1.7207169598271814
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry DEF roda Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(2)"
        boundingObject USE roda
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 26.15995899933815
        anchor 0.0744444 0.06 6.34548e-06
      }
      device [
        RotationalMotor {
          name "roda3"
        }
      ]
      endPoint Solid {
        translation 0.07444437870578856 0.06000002470010913 1.2335777097864044e-05
        rotation 0.4409561825569936 0.7817378319068861 0.44095760253145067 1.8145840051348805
        children [
          DEF roda Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry DEF roda Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(1)"
        boundingObject USE roda
        physics DEF physics_roda Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 13.35999492083436
        anchor 0.0744602 -0.0695859 -3.08998e-06
      }
      device [
        RotationalMotor {
          name "roda4"
        }
      ]
      endPoint Solid {
        translation 0.07446020933125468 -0.06958591147017797 -5.802938065182086e-06
        rotation 0.6795053793016115 0.27799017642518314 0.6789653167218989 2.5988985140737615
        scale 0.958062 0.958062 0.958062
        children [
          DEF roda Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        boundingObject USE roda
        physics DEF phisics_roda Physics {
        }
      }
    }
    DEF papan Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.2 0.05
      }
    }
  ]
  boundingObject USE papan
  physics Physics {
  }
  controller "mobil"
  linearVelocity -1.4965398540387376e-08 3.875710230340729e-11 3.6169790508616965e-10
  angularVelocity -1.999940206264476e-08 3.0149710787414436e-08 -1.9826635694002906e-11
}
